client.go

```diff
@@ -0,0 +1,65 @@
+package filetypes
+
+import (
+	csvFile "github.com/cloudquery/filetypes/csv"
+	jsonFile "github.com/cloudquery/filetypes/json"
+)
+
+type Client struct {
+	spec                   *FileSpec
+	csv                    *csvFile.Client
+	json                   *jsonFile.Client
+	csvTransformer         *csvFile.Transformer
+	csvReverseTransformer  *csvFile.ReverseTransformer
+	jsonTransformer        *jsonFile.Transformer
+	jsonReverseTransformer *jsonFile.ReverseTransformer
+}
+
+// NewClient creates a new client for the given spec
+func NewClient(spec *FileSpec) (*Client, error) {
+	err := spec.UnmarshalSpec()
+	if err != nil {
+		return &Client{}, err
+	}
+
+	spec.SetDefaults()
+	if err := spec.Validate(); err != nil {
+		return &Client{}, err
+	}
+
+	switch spec.Format {
+	case FormatTypeCSV:
+		opts := []csvFile.Options{
+			csvFile.WithDelimiter([]rune(spec.csvSpec.Delimiter)[0]),
+		}
+		if spec.csvSpec.IncludeHeaders {
+			opts = append(opts, csvFile.WithHeader())
+		}
+
+		client, err := csvFile.NewClient(opts...)
+		if err != nil {
+			return &Client{}, err
+		}
+		return &Client{
+			spec:                  spec,
+			csvTransformer:        &csvFile.Transformer{},
```

Comment:
thought:
1. make value receiver instead of pointer receiver for `csvFile.Transformer{}`
2. drop reverse transformer func from `csvFile.Transformer{}` as it should be inherited from embedded struct
3. you can change the filed to be value here and not init it al all

Reply:
same for json, btw